src/components/inputs/ui-select-extended/ui-select-extended.component.tsx

```diff
@@ -53,6 +60,16 @@ const UiSelectExtended: React.FC<FormFieldProps> = ({ question, handler, onChang
     setFieldValue(question.id, value);
     onChange(question.id, value, setErrors, setWarnings);
     handler?.handleFieldSubmission(question, value, encounterContext);
+    // check only if workflow
+    if (question.type === 'programWorkflow') {
+      setWorkflowManager((prevState) => ({
+        ...prevState,
+        [question.id]: {
+          label: question.id,
+          uuid: value,
+        },
+      }));
+    }
```

Comment:
What role does the workflow manager play?

Reply:
It stores all workflowUuids in a form and passes the appropriate `workFlowUUid` to the state based this :
"questionOptions": {
              "rendering": "ui-select-extended",
              "datasource": {
                "name": "program_workflow_state_datasource",
                "config": {
                  "associatedWorkflowId": "tbWorkflow"
                }
              },

            },